torchgeo/samplers/single.py

```diff
@@ -119,8 +126,9 @@ def __iter__(self) -> Iterator[BoundingBox]:
             (minx, maxx, miny, maxy, mint, maxt) coordinates to index a dataset
         """
         for _ in range(len(self)):
-            # Choose a random tile
-            hit = random.choice(self.hits)
+            # Choose a random tile, weighted by area
+            idx = torch.multinomial(self.areas, 1)
```

Comment:
this defaults to sampling _without_ replacement

Reply:
We only sample a single idx at a time so replacement wouldn't affect anything.